docs/design/Architecture and Requirement.md

```diff
@@ -0,0 +1,336 @@
+   
+5. It was hard to build a data pipeline which flows/transform data from raw data source to whatever required by ML applications. 
+   **Answer to that:** Open source big data industry plays an important role to provide, simplify, unify processes and building blocks for data flows, transformations, etc.
+   
+Machine learning industry is moving on the right track to solve major roadblocks. So what is the pain points now for companies who have machine learning needs? What we can help here? To answer this question, let's look at machine learning workflow first. 
+
+## Machine Learning Workflows & Pain points
+
+```
+1) From different data source such as edge, clickstream, logs, etc.
+   => Land to data lakes  
+   
+2) From data lake, data transformation: 
+   => Data transformations: Cleanup, remove invalid rows/columns, 
+                            select columns, sampling, split train/test
+                            data-set, join table, etc.
+   => Data prepared for training.
+                            
+3) From prepared data: 
+   => Training, model hyper-parameter tuning, cross-validation, etc. 
+   => Models saved to storage. 
+   
+4) From saved models: 
+   => Model assurance, deployment, A/B testing, etc.
+   => Model deployed for online serving or offline scoring.
+```
+
+Typically data scientists responsible for item 2)-4), 1) typically handled by a different team (called Data Engineering team in many companies, some Data Engineering team also responsible for part of data transformation)
+
+### Pain \#1 Complex workflow/steps from raw data to model, many different tools need by different steps, hard to make changes to workflow, and not error-proof
+
+It is a complex workflow from raw data to usable models, after talking to many different data scientists, we have learned that a typical procedure to train a new model and push to production can take months to 1-2 years. 
+
+It is also a wide skill set required by this workflow. For example, data transformation needs tools like Spark/Hive for large scale and tools like Pandas for small scale. And model training needs to be switched between XGBoost, Tensorflow, Keras, PyTorch. Building a data pipeline needs Apache Airflow or Oozie. 
+
+Yes, there are great, standardized open-source tools built for many of such purposes. But how about changes need to be made for a particular part of the data pipeline? How about adding a few columns to the training data for experiments? How about training models, and push models to validation, A/B testing before rolling to production? All these steps need jumping between different tools, UIs, and very hard to make changes, and it is not error-proof during these procedures.
+
+#### Pain \#2 Dependencies of underlying resource management platform
```

Comment:
Document heading level error.
Need change `#### Pain` to `### Pain`

Reply:
updated